src/main/java/com/metamx/common/parsers/JSONPathParser.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.metamx.common.parsers;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.spi.json.JacksonJsonProvider;
+import com.jayway.jsonpath.spi.json.JsonProvider;
+import com.jayway.jsonpath.spi.mapper.JacksonMappingProvider;
+import com.jayway.jsonpath.spi.mapper.MappingProvider;
+import com.metamx.common.Pair;
+
+import java.util.EnumSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class JSONPathParser implements Parser<String, Object>
+{
+  private Map<String, Pair<FieldType, JsonPath>> fieldPathMap;
+  private List<FieldSpec> fieldSpecs;
+  private final boolean useFieldDiscovery;
+
+  public JSONPathParser(List<FieldSpec> fieldSpecs, boolean useFieldDiscovery)
+  {
+    this.fieldSpecs = fieldSpecs;
+    this.fieldPathMap = generateFieldPaths(fieldSpecs);
+    this.useFieldDiscovery = useFieldDiscovery;
+    setJacksonProvider();
+  }
+
+  @Override
+  public List<String> getFieldNames()
+  {
+    return null;
+  }
+
+  @Override
+  public void setFieldNames(Iterable<String> fieldNames)
+  {
+  }
+
+  @Override
+  public Map<String, Object> parse(String input)
```

Comment:
There should be a try/catch here that throws ParseExceptions if there is a problem with parsing an input.

You'd be forgiven for not knowing to do, since it's not documented anywhere :P. We should update the documentation of `Parser` too.